crates/rpcsrv/src/flight/handler.rs

```diff
@@ -140,11 +99,56 @@ impl FlightSessionHandler {
             .map_err(Status::from);
         Ok(Response::new(Box::pin(stream)))
     }
+
+    pub fn new(engine: &Arc<Engine>) -> Self {
+        Self {
+            engine: engine.clone(),
+            logical_plans: DashMap::new(),
+            sessions: DashMap::new(),
+        }
+    }
+
+    async fn get_or_create_ctx<T>(
+        &self,
+        request: &Request<T>,
+    ) -> Result<Arc<Mutex<Session>>, Status> {
+        let remote = request.remote_addr().unwrap();
+
+        let ip = remote.ip().to_string();
+        let port = remote.port().to_string();
+        let conn_key = ConnKey { ip, port };
+
+        if self.sessions.contains_key(&conn_key) {
+            let sess = self.sessions.get(&conn_key).unwrap().clone();
+            return Ok(sess);
+        }
+
+        let db_id = request
+            .metadata()
+            .get(DATABASE_HEADER)
+            .and_then(|s| Uuid::try_parse_ascii(s.as_bytes()).ok())
+            .unwrap_or_else(Uuid::new_v4);
```

Comment:
This shouldn't be returning a new uuid ever.

Reply:
so we just shouldn't set the `db_id` if its not available?

Reply:
If it's not available, we should be erroring.

If we're worried about the local case, this should be the nil uuid and not a random one.

Reply:
> If we're worried about the local case, this should be the nil uuid and not a random one.

got it. That's the case I was trying to handle with the `new`. Will just set it to `nil` instead.